### PR TITLE
[sc-4942] add aws cli version validation

### PR DESCRIPTION
**Pull request by maxlevinps**

### CODE REVIEW

**Review by steb-veksler — CHANGES_REQUESTED** [review]

File: pkg/k8s/cluster.go [hunk 1]
```diff
@@ -206,6 +224,64 @@ func (clusterRequirements ClusterRequirements) validateAuthorization(ctx context
...
+func ValidateAwsCliVersionSupported(version string) bool {
+	// aws version format: aws-cli/2.7.32 Python/3.9.11 Linux/5.11.0-1021-aws exe/x86_64.ubuntu.20 prompt/off
+	// we want to extract the version number from the first part of the version string using regex
+	awsCliVersionRegex := regexp.MustCompile(`aws-cli/(\d+\.\d+\.\d+)`)
+	matches := awsCliVersionRegex.FindStringSubmatch(version)
+	if len(matches) != 2 {
+		return false
+	}
+
+	awsCliVersion, err := semver.Parse(matches[1])
```
Comment (steb-veksler) [inline pkg/k8s/cluster.go:271]: switch awsCliVersion.Major {
	case 1:
		return awsCliVersion.GTE(MinSupportedAwsCliV1Version)
	case 2:
		return awsCliVersion.GTE(MinSupportedAwsCliV2Version)
	default:
		return true
	}

File: pkg/k8s/cluster.go [hunk 2]
```diff
@@ -206,6 +224,64 @@ func (clusterRequirements ClusterRequirements) validateAuthorization(ctx context
...
+		}
+	}
+
+	return Requirement{
+		Message:      CLUSTER_CLI_AUTH_SUPPORTED,
+		IsCompatible: true,
+	}
+}
+
+func ValidateAwsCliVersionSupported(version string) bool {
```
Comment (steb-veksler) [inline pkg/k8s/cluster.go:263]: private method

File: pkg/k8s/cluster.go [hunk 3]
```diff
@@ -17,12 +18,21 @@ const (
...
+
+	INSTALL_AWS_CLI_HINT = `Hint: 
+  * Install aws cli by following https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html`
 )
 
 var (
 	GKE_CLUSTER_REGEX = regexp.MustCompile("^gke_(?P<project>.+)_(?P<zone>.+)_(?P<name>.+)$")
 	EKS_CLUSTER_REGEX = regexp.MustCompile("^arn:aws:eks:(?P<region>.+):(?P<account>.+):cluster/(?P<name>.+)$")
 
+	AwsCliV2                    = semver.Version{Major: 2, Minor: 0, Patch: 0}
```
Comment (steb-veksler) [inline pkg/k8s/cluster.go:31]: remove

File: pkg/k8s/cluster.go [hunk 4]
```diff
@@ -17,12 +18,21 @@ const (
...
+	AwsCliV3                    = semver.Version{Major: 3, Minor: 0, Patch: 0}
```
Comment (steb-veksler) [inline pkg/k8s/cluster.go:32]: remove

**Review by steb-veksler — CHANGES_REQUESTED** [review]

**Review by noamApps — APPROVED** [review]

**Pull request merged**